src/markup.ts

```diff
@@ -0,0 +1,303 @@
+  }
+
+  static keyboard(
+    buttons: HideableKBtn[] | HideableKBtn[][],
+    options?: Partial<KeyboardBuildingOptions<HideableKBtn>>
+  ) {
+    const keyboard = buildKeyboard(buttons, {
+      columns: 1,
+      ...options,
+    })
+    return new Markup({ keyboard })
+  }
+
+  static inlineKeyboard(
+    buttons: HideableIKBtn[] | HideableIKBtn[][],
+    options?: Partial<KeyboardBuildingOptions<HideableIKBtn>>
+  ) {
+    const inlineKeyboard = buildKeyboard(buttons, {
+      columns: buttons.length,
+      ...options,
+    })
+    return new Markup({ inline_keyboard: inlineKeyboard })
+  }
+
+  static button(
+    text: string,
+    hide = false
+  ): Hideable<KeyboardButton.CommonButton> {
+    return { text: text, hide: hide }
+  }
+
+  static contactRequestButton(
+    text: string,
+    hide = false
+  ): Hideable<KeyboardButton.RequestContactButton> {
+    return { text: text, request_contact: true, hide: hide }
+  }
+
+  static locationRequestButton(
+    text: string,
+    hide = false
+  ): Hideable<KeyboardButton.RequestLocationButton> {
+    return { text: text, request_location: true, hide: hide }
+  }
+
+  static pollRequestButton(
+    text: string,
+    type?: 'quiz' | 'regular',
+    hide = false
+  ): Hideable<KeyboardButton.RequestPollButton> {
+    return { text: text, request_poll: { type }, hide: hide }
+  }
+
+  static urlButton(
+    text: string,
+    url: string,
+    hide = false
+  ): Hideable<InlineKeyboardButton.UrlButton> {
+    return { text: text, url: url, hide: hide }
+  }
+
+  static callbackButton(
+    text: string,
+    data: string,
+    hide = false
+  ): Hideable<InlineKeyboardButton.CallbackButton> {
+    return { text: text, callback_data: data, hide: hide }
+  }
+
+  static switchToChatButton(
+    text: string,
+    value: string,
+    hide = false
+  ): Hideable<InlineKeyboardButton.SwitchInlineButton> {
+    return { text: text, switch_inline_query: value, hide: hide }
+  }
+
+  static switchToCurrentChatButton(
+    text: string,
+    value: string,
+    hide = false
+  ): Hideable<InlineKeyboardButton.SwitchInlineCurrentChatButton> {
+    return { text: text, switch_inline_query_current_chat: value, hide: hide }
+  }
+
+  static gameButton(
+    text: string,
+    hide = false
+  ): Hideable<InlineKeyboardButton.GameButton> {
+    return { text: text, callback_game: {}, hide: hide }
+  }
+
+  static payButton(
+    text: string,
+    hide = false
+  ): Hideable<InlineKeyboardButton.PayButton> {
+    return { text: text, pay: true, hide: hide }
+  }
+
+  static loginButton(
+    text: string,
+    url: string,
+    opts: {
+      forward_text?: string
+      bot_username?: string
+      request_write_access?: boolean
+    } = {},
+    hide = false
+  ): Hideable<InlineKeyboardButton.LoginButton> {
+    return {
+      text: text,
+      login_url: { ...opts, url: url },
+      hide: hide,
+    }
+  }
+
+  static formatHTML(text = '', entities: MessageEntity[] = []) {
```

Comment:
This function feels out of place here, how about moving it to `core/replicators`?

Reply:
Ideally, the file should export only `removeKeyboard`, `forceReply`, `keyboard` and `inlineKeyboard`.

Reply:
I factored out this and other helper methods to `core/helpers`. I also took the freedom to force push to clean up the history a bit.